src/ResourceManager/Automation/Commands.Automation/Cmdlet/UpdateManagement/NewAzureAutomationSoftwareUpdateConfiguration.cs

```diff
@@ -0,0 +1,121 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Globalization;
+
+namespace Microsoft.Azure.Commands.Automation.Cmdlet.UpdateManagement
+{
+    using System;
+    using System.Management.Automation;
+    using System.Security.Permissions;
+    using Microsoft.Azure.Commands.Automation.Common;
+    using Models = Microsoft.Azure.Commands.Automation.Model;
+    using Microsoft.Azure.Commands.Automation.Model.UpdateManagement;
+    using System.Linq;
+    using Properties;
+
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "AutomationSoftwareUpdateConfiguration",
+        SupportsShouldProcess = true)]
```

Comment:
Needs `DefaultParameterSetName = ` whichever is your default parameter set. Because, you need to choose either Windows or Linux, since neither is not a valid parameter set.

Reply:
I am setting the Default parameter set as Windows. @finiteattractor do you see any issues with that?